src/shared/Core/Interop/MacOS/MacOSEnvironment.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using GitCredentialManager.Interop.Posix;
+
+namespace GitCredentialManager.Interop.MacOS
+{
+    public class MacOSEnvironment : PosixEnvironment
+    {
+        public MacOSEnvironment(IFileSystem fileSystem)
+            : this(fileSystem, GetCurrentVariables()) { }
+
+        internal MacOSEnvironment(IFileSystem fileSystem, IReadOnlyDictionary<string, string> variables)
+            : base(fileSystem)
+        {
+            EnsureArgument.NotNull(variables, nameof(variables));
+            Variables = variables;
+        }
+
+        public override bool TryLocateExecutable(string program, out string path)
+        {
+            string homebrewPrefix = Environment.GetEnvironmentVariable("HOMEBREW_PREFIX");
+            List<string> pathsToIgnore = new List<string>();
+
+            if (homebrewPrefix != null)
+            {
+                pathsToIgnore.Add($"{homebrewPrefix}/Homebrew/Library/Homebrew/shims/shared/git");
+            }
```

Comment:
Could probably just consider ignoring _any_ path which ends in `Homebrew/Library/Homebrew/shims/shared/git`. That'll save you needing to have `HOMEBREW_PREFIX` be present or passed.

Reply:
I'm going to roll with the `-E` change Matthew suggested above. Can't hurt to have access to all the `HOMEBREW` variables in future.